OrbitGl/PickingManager.cpp

```diff
@@ -7,26 +7,41 @@
 #include "OpenGl.h"
 #include "OrbitBase/Logging.h"
 
-PickingId PickingManager::CreatePickableId(std::weak_ptr<Pickable> a_Pickable,
-                                           BatcherId batcher_id) {
+PickingId PickingManager::GetOrCreatePickableId(
+    std::weak_ptr<Pickable> pickable, BatcherId batcher_id) {
+  CHECK(!pickable.expired());
+
   absl::MutexLock lock(&mutex_);
-  ++id_counter_;
-  PickingId id =
-      PickingId::Get(PickingType::kPickable, id_counter_, batcher_id);
-  id_pickable_map_[id_counter_] = a_Pickable;
+  uint32_t pickable_id = 0;
+  auto locked_pickable = pickable.lock();
```

Comment:
I am still kind-of uneasy with this. Maybe add a comment why this is not checked for null here? Maybe a TODO: replace this with shared_ptr<> since we cannot handle the case of expired pickable.

Reply:
As discussed off-line, we could add this in a subsequent PR